tools/dev_qa_test/dev_qa_test_runner.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+
+# Dev QA Test Runner for rules_bazel_integration_test
+# This script downloads a source tarball, injects BuildBuddy toolchain,
+# and runs Bazel commands against it using BuildBuddy RBE.
+
+set -euo pipefail
+
+bazel="${BIT_BAZEL_BINARY:-}"
+workspace_dir="${BIT_WORKSPACE_DIR:-}"
+
+tarball_url="${QA_TARBALL_URL:-}"
+strip_prefix="${QA_STRIP_PREFIX:-}"
+bazel_command="${QA_BAZEL_COMMAND:-}"
+api_key="${BB_API_KEY:-}"
+
+if [[ -z "${bazel}" ]]; then
+  echo >&2 "ERROR: BIT_BAZEL_BINARY not set"
+  exit 1
+fi
+
+if [[ -z "${workspace_dir}" ]]; then
+  echo >&2 "ERROR: BIT_WORKSPACE_DIR not set"
+  exit 1
+fi
+
+if [[ -z "${tarball_url}" ]]; then
+  echo >&2 "ERROR: QA_TARBALL_URL not set"
+  exit 1
+fi
+
+if [[ -z "${strip_prefix}" ]]; then
+  echo >&2 "ERROR: QA_STRIP_PREFIX not set"
+  exit 1
+fi
+
+if [[ -z "${bazel_command}" ]]; then
+  echo >&2 "ERROR: QA_BAZEL_COMMAND not set"
+  exit 1
+fi
+
+echo "=================================================="
+echo "Dev QA Test Runner"
+echo "=================================================="
+echo "Bazel binary: ${bazel}"
+echo "Workspace: ${workspace_dir}"
+echo "Tarball URL: ${tarball_url}"
+echo "Strip prefix: ${strip_prefix}"
+echo "Bazel command: ${bazel_command}"
+echo "=================================================="
+
+cd "${workspace_dir}"
+
+if ! curl -L "${tarball_url}" -o repo.tar.gz; then
+  echo >&2 "ERROR: Failed to download tarball from ${tarball_url}"
+  exit 1
+fi
+
+if ! tar -xzf repo.tar.gz; then
+  echo >&2 "ERROR: Failed to extract tarball"
+  exit 1
+fi
+
+if [[ ! -d "${strip_prefix}" ]]; then
+  echo >&2 "ERROR: Expected directory ${strip_prefix} not found after extraction"
+  ls -la
+  exit 1
+fi
+
+cd "${strip_prefix}"
+echo "Working directory: $(pwd)"
+
+if [[ ! -f MODULE.bazel ]]; then
+  echo >&2 "ERROR: No MODULE.bazel found. Only bzlmod projects are supported."
+  exit 1
+fi
+
+echo "Detected MODULE.bazel project - injecting bazel_dep..."
+
+# Append bazel_dep and extension setup to MODULE.bazel
+cat >> MODULE.bazel <<'EOF'
+
+# BuildBuddy RBE toolchain (injected by qa_test_runner.sh)
+bazel_dep(name = "toolchains_buildbuddy", version = "0.0.2")
+
+# Use the extension to create toolchain and platform targets
+buildbuddy = use_extension("@toolchains_buildbuddy//:extensions.bzl", "buildbuddy")
+
+# Workaround: toolchains_buildbuddy loads buildifier_prebuilt but declares it as dev_dependency
```

Comment:
Good first find for the test :⁠-⁠)

Reply:
It ended up being fake news!
Pretty sure the test was modifying MODULE.bazel but invoking WORKSPACE at one point.

This is a big relief -- otherwise it would mean that the CI tests for the toolchain and the BCR presubmit failed to catch this "problem"!